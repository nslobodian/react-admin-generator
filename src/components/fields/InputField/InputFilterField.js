import React from 'react'
import compose from 'recompose/compose'
import withState from 'recompose/withState'
import withHandlers from 'recompose/withHandlers'

export const enhance = compose(
  withState('debounce', 'setDebounce', null),
  withHandlers({
    handleChange: ({ setDebounce, input: { onChange, name }, changeAttempt, debounce }) => ({ target: { value } }) => {
      onChange(value)
      clearTimeout(debounce)
      setDebounce(setTimeout(() => {
        changeAttempt(name, value)
      }, 500))
    },
  })
)

export default enhance(({ input: { value, name, onBlur }, placeholder = '', handleChange, className, type = 'text' }) =>
  <input value={value} name={name} onChange={handleChange} placeholder={placeholder} onBlur={onBlur} className={className} type={type} />
)
