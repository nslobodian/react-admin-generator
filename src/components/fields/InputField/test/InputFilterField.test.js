import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import InputFilterField from '../InputFilterField'

describe('(Components) InputFilterField', () => {
  const props = {
      input: {
        name: 'name',
        value: 1,
        onBlur: jest.fn(),
        onChange: jest.fn(),
      },
      handleChange: {
        setDebounce: {
          setTimeout: jest.fn(),
        },
        clearTimeout: jest.fn(),
        changeAttempt: jest.fn(),
      },
      className: 'className',
    }
  it('Should render component properly', () => {
    const wrapper = shallow(
      <InputFilterField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('Focus on input', () => {
    const wrapper = mount(
      <InputFilterField
        {...props}
      />)
    wrapper.find('input').simulate('focus')
  })
  it('Change on input', () => {
    const wrapper = mount(
      <InputFilterField
        {...props}
      />)
    wrapper.find('input').simulate('change')
  })
})
