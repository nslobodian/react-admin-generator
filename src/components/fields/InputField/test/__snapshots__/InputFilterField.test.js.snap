// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) InputFilterField Should render component properly 1`] = `
<_class
  className="className"
  debounce={null}
  handleChange={
    Object {
      "changeAttempt": [Function],
      "clearTimeout": [Function],
      "setDebounce": Object {
        "setTimeout": [Function],
      },
    }
  }
  input={
    Object {
      "name": "name",
      "onBlur": [Function],
      "onChange": [Function],
      "value": 1,
    }
  }
  setDebounce={[Function]}
/>
`;
